DedicatedServer/Scripts/Libs/Zrx/ModeLibs/FlagRush/FlagRush_RpcEvents.Script.txt

```diff
@@ -0,0 +1,280 @@
+#Const 		C_XMLRPC_WarmUpEnd 															"Trackmania.WarmUp.End"
+
+
+#Struct K_Team {
+	Integer Num;
+	Text Name;
+}
+
+#Struct K_RpcScore {
+	Integer MatchPoints;
+	Integer FlagsScored;
+	Integer FlagsStealed;
+	Integer FlagsAssists;
+}
+
+// Structs
+#Struct K_RpcPlayer {
+	Text Login;
+	Text Name;
+	Integer Team;
+	Text TeamName;
+	Vec3 Position;
+}
+
+#Struct K_RpcPlayerScore {
+	Text Login;
+	Text Name;
+	Integer Team;
+	Text TeamName;
+	K_RpcScore RoundScore;
+	K_RpcScore MapScore;
+	K_RpcScore MatchScore;
+}
+
+#Struct K_RpcTeam {
+	Integer Id;
+	Text Name;
+	Integer RoundScore;
+	Integer MapScore;
+	Integer MatchScore;
+}
+
+
+#Struct K_RpcFlag {
+	Vec3 Position;
+	K_RpcPlayer Carrier;
+}
+
+#Struct K_RpcMap {
+	Text AuthorLogin;
+	Text AuthorName;
+	Text Name;
+}
+
+
+#Struct K_RpcCbCarrierFlag {
+	Integer Time;
+	K_RpcPlayer OldCarrier;
+	K_RpcFlag Flag;
+}
+
+#Struct K_RpcCbPlayer {
+	Integer Time;
+	K_RpcPlayer Player;
+}
+
+#Struct K_RpcCbScoreFlag {
+	Integer Time;
+	K_RpcPlayer AssistPlayer;
+	K_RpcPlayer Player;
+	K_RpcTeam Team1Score;
+	K_RpcTeam Team2Score;
+}
+
+#Struct K_RpcCbTime {
+	Integer Time;
+}
+
+#Struct K_RpcCbFlow {
+	Integer		Time;
+	K_RpcMap 	Map;
+	K_RpcTeam Team1Score;
+	K_RpcTeam Team2Score;
+}
+
+#Struct K_RpcCbPlayerFlow {
+	Integer		Time;
+	K_RpcMap 	Map;
+	K_RpcPlayerScore[] Players;
+	K_RpcTeam Team1Score;
+	K_RpcTeam Team2Score;
+}
+
+
+declare Integer G_MatchStartTime;
+
+Void Init() {
+	G_MatchStartTime = Now;
+	XmlRpc::RegisterCallback(C_XMLRPC_FlagPickup, "Called when player picks up the flag");
+	XmlRpc::RegisterCallback(C_XMLRPC_FlagSteal, "Called when flag is stealed");
+	XmlRpc::RegisterCallback(C_XMLRPC_FlagDrop, 	"Called when flag is droped");
+	XmlRpc::RegisterCallback(C_XMLRPC_FlagScored, "Called when flag is scored");
+	XmlRpc::RegisterCallback(C_XMLRPC_FlagReset, "Called when flag is reset");
+	XmlRpc::RegisterCallback(C_XMLRPC_Overtime, "Called when flag is reset");
+	XmlRpc::RegisterCallback(C_XMLRPC_PlayerDeath, "Called when player gets out of bounds");
+
+	XmlRpc::RegisterCallback(C_XMLRPC_MatchStart, "Called when event occurs");
+	XmlRpc::RegisterCallback(C_XMLRPC_MatchEnd, "Called when event occurs");
+	XmlRpc::RegisterCallback(C_XMLRPC_MapStart, "Called when event occurs");
+	XmlRpc::RegisterCallback(C_XMLRPC_MapEnd, "Called when event occurs");
+	XmlRpc::RegisterCallback(C_XMLRPC_RoundStart, "Called when event occurs");
+	XmlRpc::RegisterCallback(C_XMLRPC_RoundEnd, "Called when event occurs");
+}
+
+Void SetMatchStart() {
+	G_MatchStartTime = Now;
+}
+
+Integer GetMatchTime() {
+	return Now - G_MatchStartTime;
+}
+
+K_Team GetTeamFromLogin(Text Login) {
+	declare CSmPlayer Player <=> GetPlayer(Login);
+	declare CTeam Team;
+	if (Player == Null) return K_Team{};
+	if (Player.CurrentClan > 0) Team <=> Teams[Player.CurrentClan - 1];
+	if (Team == Null) return K_Team{};
+	return K_Team{Num = Player.CurrentClan, Name = Team.Name };
+}
```

Comment:
I don't see the point of using a K_Team struct here at all. Instead just return the actual Team.

```suggestion
CTeam GetTeamFromLogin(Text Login) {
	declare CSmPlayer Player <=> GetPlayer(Login);
	if (Player == Null || !Teams.existskey(Player.CurrentClan - 1)) return Null;
	return Teams[Player.CurrentClan - 1];
}
```